server_installation/topics/operating-mode/crossdc/proc-setting-up-infinispan.adoc

```diff
@@ -0,0 +1,172 @@
++
+By default, {jdgserver_name} Server uses `server/conf/infinispan.xml` for static configuration such as cluster transport and security mechanisms. 
+
+. Create a stack the uses TCCPING. The default tcp stack finds nodes by using IP multicast, which is typically unavailable.
++
+[source,xml,options="nowrap",subs=attributes+]
+----
+<stack name="global-cluster" extends="tcp">
+    <!-- Remove MPING protocol from the stack and add TCPPING -->
+    <TCPPING initial_hosts="<ip1>[<port>],<ip2>[<port>]" stack.combine="REPLACE" stack.position="MPING"/>
+</stack>
+----
++
+NOTE: The `initial_hosts` file contains the list of all Infinispan servers at both sites.
+
+. Configure the {jdgserver_name} cluster transport to perform Cross-Datacenter replication.
+.. Add the RELAY2 protocol to a JGroups stack.
++
+[source,xml,options="nowrap",subs=attributes+]
+----
+<jgroups>
+   <stack name="xsite" extends="udp"> <1>
+      <relay.RELAY2 site="site1" <2>
+                    max_site_masters="1000"/> <3>
+      <remote-sites default-stack="global-cluster"> <4>
+         <remote-site name="site1"/>
+         <remote-site name="site2"/>
+      </remote-sites>
+   </stack>
+</jgroups>
+----
+<1> Creates a stack named `xsite` that extends the default UDP cluster transport.
+<2> Adds the RELAY2 protocol and names the cluster you are configuring as `site1`. +
+The site name must be unique to each {jdgserver_name} cluster.
+<3> Sets 1000 as the number of relay nodes for the cluster. You should set a value that is equal to or greater than the maximum number of nodes in your {jdgserver_name} cluster.
+<4> Names all {jdgserver_name} clusters that backup caches with {jdgserver_name} data and uses the default TCP stack for inter-cluster transport.
++
+.. Configure the {jdgserver_name} cluster transport to use the stack.
++
+[source,xml,options="nowrap",subs=attributes+]
+----
+<cache-container name="default" statistics="true">
+      <transport cluster="${infinispan.cluster.name:cluster}"
+                 stack="xsite"/> <1>
+</cache-container>
+----
+<1> Uses the `xsite` stack for the cluster.
++
+. Configure the keystore as an SSL identity in the server security realm.
++
+[source,xml,options="nowrap",subs=attributes+]
+----
+<server-identities>
+  <ssl>
+    <keystore path="server.jks" <1>
+              relative-to="infinispan.server.config.path"
+              keystore-password="password" <2>
+              alias="server" /> <3>
+  </ssl>
+</server-identities>
+----
+<1> Specifies the path of the keystore that contains the SSL identity.
+<2> Specifies the password to access the keystore.
+<3> Names the alias of the certificate in the keystore.
++
+. Configure the authentication mechanism for the Hot Rod endpoint.
++
+[source,xml,options="nowrap",subs=attributes+]
+----
+<endpoints socket-binding="default" security-realm="default">
+   <hotrod-connector name="hotrod">
+      <authentication security-realm="default">
+         <sasl mechanisms="SCRAM-SHA-512" <1>
```

Comment:
I am guessing at the coding of this line.  Can you correct? @mposolda

Reply:
I'll let Marek provide the definitive answer as you're asking him but that looks right to me. I added it to a server config to double check too. One little nit, I don't think it's necessary to add `security-realm="default"` to the authentication element as well because that is defined globally for the `endpoints` element and the Hot Rod connector will use that.

Reply:
this looks good to me now Andy. thanks.